types/scenegraph.d.ts

```diff
@@ -10,6 +10,122 @@ declare interface ScaleFactor {
+
+ // Later on...
+ // (This could be in a different plugin, if it passes the original plugin's ID here)
+ let jsonString = node.sharedPluginData.getItem(PLUGIN_ID, "richData");
+ if (jsonString) {  // may be undefined
+    let richObjectCopy = JSON.parse(jsonString);
+    console.log(richObjectCopy.list.length);  // 3
+ }
+ */
+declare interface PerPluginStorage {
+    /**
+     * Returns a map where key is plugin ID and value is a nested map containing all the shared metadata for that plugin ID (i.e. the result of calling `getForPluginId()` with that ID).
+     */
+    getAll(): { [key: string]: {[key:string]: string} };
+
+    /**
+     * Returns a map of key-value string pairs containing all shared metadata stored on this node by the given plugin. May be an empty object (zero keys), but is never null.
+     *
+     * This map is a clone of the stored metadata, so modifying it has no effect.
+     * @param pluginId
+     *
+     * @example
+     *const MY_PLUGIN_ID = "<your manifest's plugin ID here>";
+let mySharedMetadata = node.sharedPluginData.getForPluginId(MY_PLUGIN_ID);
+console.log("My shared 'foo' & 'bar' values:",
+    mySharedMetadata.foo, mySharedMetadata.bar);
+
+
+console.log("Plugin B's shared 'foo' value:",
+    node.sharedPluginData.getForPluginId("B").foo);
+     */
+    getForPluginId(pluginId: string): {[key:string]: string};
+
+    /**
+     * Returns a list of all keys stored on this node by the given plugin. May be empty (length zero), but is never null.
+     * @param pluginId
+     *
+     * @example
+     * console.log("All properties stored by plugin A on this node:",
+     node.sharedPluginData.keys("A"));
+     */
+    keys(pluginId: string): string[];
+
+    /**
+     * Returns the value stored under the given key on this node by the given plugin, or `undefined` if the plugin hasn't stored anything under the given key.
+     *
+     * Because metadata is stored separately per plugin, two plugins can store two different values under the same key.
+     * @param pluginId
+     * @param key
+     *
+     * @example
+     * // These are two different values, stored independently per plugin
+ console.log("Plugin A's 'foo' value:", node.sharedPluginData.getItem("A", "foo"));
+ console.log("Plugin B's 'foo' value:", node.sharedPluginData.getItem("B", "foo"));
+     */
+    getItem(pluginId: string, key: string): string | undefined;
+
+    /**
+     * Set a metadata key which can be read by any other plugin.
+     * @param pluginId *Must* be equal to your plugin's ID.
+     * @param key
+     * @param value If undefined, behaves as if you'd called `removeItem()` instead.
+     *
+     * @example
+const MY_PLUGIN_ID = "<your manifest's plugin ID here>";
+node.sharedPluginData.setItem(MY_PLUGIN_ID, "foo", "42");
+
+node.sharedPluginData.setItem("other_plugin_id", "foo", "42");
+// ^ ERROR: other plugin's metadata is read-only
+
+console.log(node.sharedPluginData.getItem(MY_PLUGIN_ID, "foo"));  // "42"
+     */
```

Comment:
Formatting.

```suggestion
     * @example
     * ```
     * const MY_PLUGIN_ID = "<your manifest's plugin ID here>";
     * node.sharedPluginData.setItem(MY_PLUGIN_ID, "foo", "42");
     * 
     * node.sharedPluginData.setItem("other_plugin_id", "foo", "42");
     * // ^ ERROR: other plugin's metadata is read-only
     * 
     * console.log(node.sharedPluginData.getItem(MY_PLUGIN_ID, "foo"));  // "42"
     * ```
```